### PR TITLE
RichCommand: QuickLook

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #issue...

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Install and open QuickLook
   2. Open Files and select an item
   3. Press the space bar and see if the QuickLook preview opens
   4. Select a different item with the arrow keys or mouse and see if the preview updates
   5. Select multiple items and press the space bar, verify that the items remain selected.

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/Views/LayoutModes/StandardLayoutMode.cs [hunk 1]
```diff
@@ -170,7 +170,7 @@ protected virtual async void FileList_SelectionChanged(object sender, SelectionC
 			SelectedItems = ListViewBase.SelectedItems.Cast<ListedItem>().Where(x => x is not null).ToList();
 
 			if (SelectedItems.Count == 1 && App.AppModel.IsQuickLookAvailable)
-				await QuickLookHelpers.ToggleQuickLook(ParentShellPageInstance, true);
+				await QuickLookHelpers.ToggleQuickLook(SelectedItem.ItemPath, true);
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:173]: @cinqmilleans how can I trigger the action here?

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 2]
```diff
@@ -286,14 +286,6 @@ protected override async void FileList_PreviewKeyDown(object sender, KeyRoutedEv
 				FilePropertiesHelpers.ShowProperties(ParentShellPageInstance);
 				e.Handled = true;
 			}
-			else if (e.Key == VirtualKey.Space)
-			{
-				if (!IsRenamingItem && !ParentShellPageInstance.ToolbarViewModel.IsEditModeEnabled)
-				{
-					e.Handled = true;
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:293]: This e.Handled = true is very important, it prevents the selection from being reset if multiple items are selected.

File: src/Files.App/Views/LayoutModes/StandardLayoutMode.cs (hunk 1 above)
Comment (cinqmilleans) [reply src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:173]: Declare the service in the class.
`private readonly ICommandManager commands = Ioc.Default.GetRequiredService<ICommandManager>();`

The method simply becomes this. This code checks that the action is IsExecutable, and executes it if possible.
```
protected virtual async void FileList_SelectionChanged(object sender, SelectionChangedEventArgs e)
{
    await commands.LaunchQuickLook.ExecuteAsync();
}
```

Comment (yaira2) [reply src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:177]: Can I pass an argument?

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs (hunk 2 above)
Comment (cinqmilleans) [reply src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:293]: I don't know what your problem is. Could you explain.

File: src/Files.App/Views/LayoutModes/StandardLayoutMode.cs (hunk 1 above)
Comment (hecksmosis) [reply src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:177]: No, I think you have to use contexts to get the value that you need

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs (hunk 2 above)
Comment (yaira2) [reply src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:293]: It's alright, I worked around this be keeping the code behind.

File: src/Files.App/Views/LayoutModes/StandardLayoutMode.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:177]: Changing the selection of an item (without pressing the space key) is supposed to toggle QuickLook but with a different command. I guess we'll need a separate action for this?

Comment (cinqmilleans) [reply src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:177]: No. IContentPageContext allows you to be notified when the selection changes, and to know the selection and the parent folder. The action can act accordingly.

**Comment by cinqmilleans** [comment]
There is a bug with the Space key. If you select an image file, then click on the search bar, you cannot use Space, it opens QuickLook. I will create another pr to fix this which is related to centralized hotkey management.

**Comment by yaira2** [comment]
> There is a bug with the Space key. If you select an image file, then click on the search bar, you cannot use Space, it opens QuickLook. I will create another pr to fix this which is related to centralized hotkey management.

Does this affect the ctrl + a shortcut as well?

**Comment by cinqmilleans** [comment]
Regarding quicklook. While testing, I had the reflex to try to close it with the Escape key. Maybe add this hotkey to exit. (Command quit/exit, which can close active popup, settings, quicklook, properties,... ). It's another pr but I note the idea.

**Comment by yaira2** [comment]
> Regarding quicklook. While testing, I had the reflex to try to close it with the Escape key. Maybe add this hotkey to exit. (Command quit/exit, which can close active popup, settings, quicklook, properties,... ). It's another pr but I note the idea.

Good idea 👍

**Pull request merged**